Description: Null-terminate a call to execlp().
Forwarded: yes
Author: Peter Pentchev <pp@storpool.com>
Last-Update: 2017-07-30

--- a/txn-install.c
+++ b/txn-install.c
@@ -994,7 +994,7 @@
 		    asprintf(&mode, "%o", orig_sb.st_mode & 03777) < 0)
 			err(1, "Could not allocate memory for the recreated file's attributes");
 
-		execlp("install", "install", "-c", "-o", owner, "-g", group, "-m", mode, "--", temp_filename, filename);
+		execlp("install", "install", "-c", "-o", owner, "-g", group, "-m", mode, "--", temp_filename, filename, NULL);
 		err(1, "Could not execute 'install' to recreate '%s'", filename);
 	}
 
