Description: Specify a mode with open(O_CREAT).
Forwarded: yes
Author: Peter Pentchev <pp@storpool.com>
Last-Update: 2017-07-29

--- a/txn-install.c
+++ b/txn-install.c
@@ -232,7 +232,7 @@
 		}
 	}
 
-	const int fd = open(idx, O_CREAT | O_EXCL | O_RDWR);
+	const int fd = open(idx, O_CREAT | O_EXCL | O_RDWR, 0644);
 	if (fd == -1)
 		err(1, "Could not create the database index '%s'", idx);
 	if (!writen(fd, INDEX_FIRST, INDEX_NUM_SIZE + 1))
@@ -568,7 +568,7 @@
 		warn("Could not generate a patch filename for '%s'", dst);
 		return (false);
 	}
-	const int patch_fd = open(patch_filename, O_RDWR | O_CREAT | O_EXCL);
+	const int patch_fd = open(patch_filename, O_RDWR | O_CREAT | O_EXCL, 0600);
 	if (patch_fd == -1) {
 		warn("Could not create the '%s' patch file for '%s'", patch_filename, dst);
 		return (false);
@@ -740,7 +740,7 @@
 	char *backup_filename;
 	if (asprintf(&backup_filename, "%s/txn.%06zu", db.dir, ln.idx) < 0)
 		err(1, "Could not generate a patch filename for '%s'", fname);
-	const int backup_fd = open(backup_filename, O_RDWR | O_CREAT | O_EXCL);
+	const int backup_fd = open(backup_filename, O_RDWR | O_CREAT | O_EXCL, 0600);
 	if (backup_fd == -1)
 		err(1, "Could not create the '%s' patch file for '%s'", backup_filename, fname);
 	if (flock(backup_fd, LOCK_EX | LOCK_NB) == -1)
